Tests/Domain/OptionalDateTimeTests.cs

```diff
@@ -30,9 +31,9 @@ private OptionalDateTime Date(int? year = null, int? month = null, int? day = nu
 			var second = Date(secondYear, secondMonth, secondDay);
 
 			if (equals)
-				Assert.AreEqual(first, second, "Dates are equal");
+				second.As<object>()/* Casting is required. */.Should().Be(first, "Dates are equal");
 			else
-				Assert.AreNotEqual(first, second, "Dates are not equal");
+				second.As<object>()/* Casting is required. */.Should().NotBe(first, "Dates are not equal");

```

Comment:
Casting is required, because the `OptionalDateTime` class overrides the `Equals` method.